src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -415,7 +503,9 @@ private bool Equals(TupleTypeSymbol other, bool ignoreCustomModifiers, bool igno
 
                 for (int i = 0; i < count; i++)
                 {
-                    if (fields[i].Name != otherFields[i].Name)
+                    if ((object)fields[i] == null ||
+                        (object)otherFields[i] == null ||
+                        fields[i].Name != otherFields[i].Name)
```

Comment:
The reason I added this code is to handle when one side is null. I'll stop by to discuss whether null check is necessary.